server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -143,8 +158,19 @@ void run() {
 
                 logger.debug("Reconciliation is complete");
 
-                desiredBalanceMetrics.updateMetrics(allocationStats, desiredBalance.weightsPerNode());
+                updateDesireBalanceMetrics(allocationStats);
+            }
+        }
+
+        private void updateDesireBalanceMetrics(AllocationStats allocationStats) {
+            var stats = statsProvider.stats(desiredBalance);
+            Map<DiscoveryNode, NodeAllocationStats> nodeAllocationStats = new HashMap<>(stats.size());
+            for (var entry : stats.entrySet()) {
+                var node = allocation.nodes().get(entry.getKey());
+                assert node != null;
```

Comment:
This could be null: 
https://github.com/elastic/elasticsearch/blob/69150c8de16335e0c3444d6bf5ccb4b705d359ad/server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java#L583-L586

Reply:
pushed https://github.com/elastic/elasticsearch/pull/116585/commits/fd6650ffaba565d007338b870be69d86598e2f86